src/EnergyPlus/InputProcessing/InputProcessor.cc

```diff
@@ -1256,6 +1264,177 @@ void InputProcessor::getObjectDefMaxArgs(std::string const &ObjectWord, // Objec
+
+        // Locations in JSON schema storing the positional aspects from the IDD format, legacy prefixed
+        auto const &legacy_idd = object_schema["legacy_idd"];
+        auto const &legacy_idd_fields = legacy_idd["fields"];
+
+        // Look for extensible
+        auto key = legacy_idd.find("extension");
+        std::string extension_key;
+        if (key != legacy_idd.end()) {
+            extension_key = key.value().get<std::string>();
+        }
+
+        for (auto const &ep_object: objects) {
+
+            // Count number of objects
+            ++DataOutputs::iNumberOfRecords;
+
+            // Loop on all regular fields
+            for (size_t i = 0; i < legacy_idd_fields.size(); ++i) {
+
+                std::string const &field = legacy_idd_fields[i];
+
+                // This is weird, but some objects like Building have a Name default... and it's not in the patternProperties
+                if (has_idd_name_field && field == "name") {
+                    auto const &name_iter = schema_name_field.value();
+                    if (name_iter.find("default") != name_iter.end()) {
+                        ++DataOutputs::iTotalFieldsWithDefaults;
+                        auto it = ep_object.find(field);
+                        if (it == ep_object.end()) {
+                            ++DataOutputs::iNumberOfDefaultedFields;
+                        }
+                    }
+                    continue;
+                }
+
+                auto const &schema_field_obj = schema_obj_props[field];
+
+                processField(field, ep_object, schema_field_obj);
+
+            } // End regular fields
+
+            auto const &legacy_idd_extensibles_iter = legacy_idd.find("extensibles");
+            if (legacy_idd_extensibles_iter != legacy_idd.end()) {
+                auto const epJSON_extensions_array_itr = ep_object.find(extension_key);
+                if (epJSON_extensions_array_itr != ep_object.end()) {
+                    auto const &legacy_idd_extensibles = legacy_idd_extensibles_iter.value();
+                    auto const &epJSON_extensions_array = epJSON_extensions_array_itr.value();
+                    auto const &schema_extension_fields = schema_obj_props[extension_key]["items"]["properties"];
+
+                    for (auto it = epJSON_extensions_array.begin(); it != epJSON_extensions_array.end(); ++it) {
+                        auto const &epJSON_extension_obj = it.value();
+                        for (size_t i = 0; i < legacy_idd_extensibles.size(); ++i) {
+                            std::string const &field = legacy_idd_extensibles[i];
+                            auto const &schema_extension_field_obj = schema_extension_fields[field];
+
+                            processField(field, epJSON_extension_obj, schema_extension_field_obj);
+                        }
+                    }
+                }
+            } // End extensible fields
```

Comment:
Note that I am handling the extensible groups, which seems to be limited to a couple of objects (at least `SurfaceProperty:SurroundingSurfaces` has an extensible with a default of 0.0 for view Factor)